### ISSUE TITLE
Translation: Sanitise LLM data with structured output parsing

### ISSUE THREAD

**Issue by bryceknz** [bryceknz/sanity-translations-experiment#11]
# Description

Currently the translation functionality is reliant on the output of the LLM being in a specific format. We want to move away from this and instead use a structured output format that can easily be parsed.

Investigate usage of the following pattern for structured output: https://js.langchain.com/docs/modules/chains/popular/structured_output

If this is suitable, then update the [`translate`](https://github.com/bryceknz/sanity-translations-experiment/blob/db6530f9d455fe1d7a5bd5d31edc59aab33382c9/server/translation.ts#L13) function to use this format.

# Acceptance criteria

- [ ] [`translate`](https://github.com/bryceknz/sanity-translations-experiment/blob/db6530f9d455fe1d7a5bd5d31edc59aab33382c9/server/translation.ts#L13) function is updated to use structured output
- [ ] [`extractJSON`](https://github.com/bryceknz/sanity-translations-experiment/blob/db6530f9d455fe1d7a5bd5d31edc59aab33382c9/server/utils.ts#L9) function is removed or updated as required

**Comment by umxr** [comment]
I dont think this provides us with enough value at our current stage of development. I think its worth picking this ticket backup once we're in an almost "Go Live" position
